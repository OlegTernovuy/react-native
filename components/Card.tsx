import { FlatList, Text } from 'react-native';

import { CardItem, VerticalList } from '../styled/Card.style';
import { DATA } from '../data/index';

const Card = () => {
    return (
        <VerticalList showsVerticalScrollIndicator={false}>
            <FlatList
                data={DATA}
                renderItem={({ item }) => (
                    <CardItem key={item.id}>
                        <Text>{item.name}</Text>
                    </CardItem>
                )}
                keyExtractor={(item) => item.id}
                contentContainerStyle={{ columnGap: 16 }}
                horizontal
            />
        </VerticalList>
    );
};

export default Card;
