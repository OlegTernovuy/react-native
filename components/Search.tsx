import { Feather } from '@expo/vector-icons';
import { router } from 'expo-router';
import { useState } from 'react';

import {
    SearchButton,
    SearchContainer,
    SearchInput,
    SearchWrapper,
} from '../styled/index.style';

const Search = () => {
    const [searchTerm, setSearchTerm] = useState('');
    return (
        <SearchContainer>
            <SearchWrapper>
                <SearchInput
                    value={searchTerm}
                    onChangeText={(text) => setSearchTerm(text)}
                    placeholder="What are you looking for"
                />
            </SearchWrapper>
            <SearchButton
                onPress={() => {
                    router.push(`/search/${searchTerm}`);
                    setSearchTerm('');
                }}
                testID="search-btn"
            >
                <Feather name="search" size={24} color="black" />
            </SearchButton>
        </SearchContainer>
    );
};

export default Search;
