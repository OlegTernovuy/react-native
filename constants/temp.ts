const OCCUPATION_TYPE = ['Development', 'Testing', 'Engineering'];

const mockData = [
    {
        name: 'Raquel Powlowski',
        avatar: 'https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/672.jpg',
        occupation: 'Development',
        responsibilitiesAtWork:
            'Andy shoes are designed to keeping in mind durability as well as trends, the most stylish range of shoes & sandals',
        id: '1',
    },
];

export { OCCUPATION_TYPE, mockData };
