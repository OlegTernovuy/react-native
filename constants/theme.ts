export const THEME = {
    Colors: {
        primary: '#312651',
        secondary: '#444262',

        lightGray: '#f1f1f1cc',
        lightBlue: '#b1d0ffcc',

        lightWhite: '#ffffff',
    },
    Sizes: {
        small: '4px',
        medium: '8px',
        large: '16px',
        xLarge: '20px',
        xxLarge: '24px',
    },
};
