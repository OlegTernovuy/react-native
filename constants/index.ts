import { OCCUPATION_TYPE } from './temp';
import { THEME } from './theme';

export { OCCUPATION_TYPE, THEME };
