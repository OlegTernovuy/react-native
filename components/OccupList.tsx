import { FlatList, Text } from 'react-native';
import { router } from 'expo-router';
import { useState } from 'react';

import { OccupButton, OccupWrapper } from '../styled/Occup.style';
import { OCCUPATION_TYPE } from '../constants';

const OccupList = () => {
    const [activeOccup, setActiveOccup] = useState('');

    return (
        <OccupWrapper>
            <FlatList
                data={OCCUPATION_TYPE}
                renderItem={({ item }) => (
                    <OccupButton
                        onPress={() => {
                            setActiveOccup(item);
                            router.push(`/search/${item}`);
                        }}
                        item={item}
                        activeOccup={activeOccup}
                    >
                        <Text>{item}</Text>
                    </OccupButton>
                )}
                contentContainerStyle={{ columnGap: 8 }}
                horizontal
            />
        </OccupWrapper>
    );
};

export default OccupList;
