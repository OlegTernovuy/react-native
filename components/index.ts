import Card from './Card';
import OccupList from './OccupList';
import Search from './Search';

export { Card, OccupList, Search };
