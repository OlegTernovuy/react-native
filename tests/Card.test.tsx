import { fireEvent, render } from '@testing-library/react-native';
import { waitFor } from '@testing-library/react-native';
import MockAdapter from 'axios-mock-adapter';
import { router } from 'expo-router';
import axios from 'axios';

import { mockData } from '../constants/temp';
import { Card } from '../components';

const mock = new MockAdapter(axios);

jest.mock('expo-router', () => ({
    router: {
        push: jest.fn(),
    },
}));

describe('Card component', () => {
    afterEach(() => {
        mock.reset();
    });

    it('should matches snapshot when data is loaded', async () => {
        mock.onGet().reply(200, mockData);
        const { toJSON } = render(<Card />);
        await waitFor(() => {
            expect(toJSON()).toMatchSnapshot();
        });
    });

    it('should matches snapshot when data is loading', async () => {
        mock.onGet().reply(() => new Promise(() => {}));
        const { toJSON } = render(<Card />);
        expect(toJSON()).toMatchSnapshot();
    });

    it('should show loader indicator when data is fetching', async () => {
        mock.onGet().reply(200, mockData);
        const { getByTestId } = render(<Card />);
        expect(getByTestId('activity-indicator')).toBeTruthy();
    });

    it('should render list of cards when data is fetched', async () => {
        mock.onGet().reply(200, mockData);
        const { getByText } = render(<Card />);
        await waitFor(() => {
            expect(getByText('Raquel Powlowski')).toBeTruthy();
        });
    });

    it('should contains cards from data', async () => {
        mock.onGet().reply(200, mockData);
        const { getByText } = render(<Card />);
        await waitFor(() => {
            mockData.forEach((item) => {
                expect(getByText(item.name)).toBeTruthy();
            });
        });
    });

    it('should navigate to card details page', async () => {
        mock.onGet().reply(200, mockData);
        const { getByText } = render(<Card />);
        const card = await waitFor(() => getByText('Raquel Powlowski'));

        fireEvent.press(card);

        expect(router.push).toHaveBeenCalledWith(`/details/1`);
    });
});
