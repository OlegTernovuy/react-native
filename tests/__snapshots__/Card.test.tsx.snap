// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should matches snapshot when data is loaded 1`] = `
<ActivityIndicator
  testID="activity-indicator"
/>
`;

exports[`Card component should matches snapshot when data is loading 1`] = `
<ActivityIndicator
  testID="activity-indicator"
/>
`;
